### PR TITLE
docs: core docs migration (rst => mdx) -  2

### PR DESCRIPTION
**Pull request by rjitsu**
Child of #8048 

Files Migrated in this PR

<details>
<summary>docs/graphql/core/hasura-cli/ ✅</summary>

- [x] docs/graphql/core/hasura-cli/hasura.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_actions.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_actions_codegen.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_actions_create.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_actions_use-codegen.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_completion.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_console.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_deploy.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_init.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_metadata.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_metadata_apply.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_metadata_clear.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_metadata_diff.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_metadata_export.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_metadata_inconsistency.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_metadata_inconsistency_drop.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_metadata_inconsistency_list.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_metadata_inconsistency_status.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_metadata_reload.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_migrate.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_migrate_apply.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_migrate_create.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_migrate_delete.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_migrate_squash.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_migrate_status.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_plugins.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_plugins_install.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_plugins_list.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_plugins_uninstall.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_plugins_upgrade.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_scripts.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_scripts_update-project-v2.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_scripts_update-project-v3.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_seed.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_seed_apply.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_seed_create.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_update-cli.rst ✅
- [x] docs/graphql/core/hasura-cli/hasura_version.rst ✅
- [x] docs/graphql/core/hasura-cli/index.rst ✅
- [x] docs/graphql/core/hasura-cli/install-hasura-cli.rst ✅
- [x] docs/graphql/core/hasura-cli/uninstall-hasura-cli.rst ✅
</details>

<details>
<summary>docs/graphql/core/migrations/ ✅</summary>

- [x] docs/graphql/core/migrations/index.rst ✅
- [x] docs/graphql/core/migrations/manage-metadata.rst ✅
- [x] docs/graphql/core/migrations/migrations-setup.rst ✅
- [x] docs/graphql/core/migrations/upgrade-v3.rst ✅
- [x] docs/graphql/core/migrations/advanced ✅
  - [x] docs/graphql/core/migrations/advanced/auto-apply-migrations.rst ✅
  - [x] docs/graphql/core/migrations/advanced/index.rst ✅
  - [x] docs/graphql/core/migrations/advanced/resetting-migrations.rst ✅
  - [x] docs/graphql/core/migrations/advanced/rolling-back-migrations.rst ✅
  - [x] docs/graphql/core/migrations/advanced/seed-data-migration.rst ✅
  - [x] docs/graphql/core/migrations/advanced/writing-migrations-manually.rst ✅
- [x] docs/graphql/core/migrations/config-v1/ ✅
  - [x] docs/graphql/core/migrations/config-v1/index.rst ✅
  - [x] docs/graphql/core/migrations/config-v1/reference/ ✅
      - [x] docs/graphql/core/migrations/config-v1/reference/index.rst ✅
      - [x] docs/graphql/core/migrations/config-v1/reference/metadata-file-format.rst ✅
      - [x] docs/graphql/core/migrations/config-v1/reference/migration-file-format.rst ✅
- [x] docs/graphql/core/migrations/config-v2/ ✅
  - [x] docs/graphql/core/migrations/config-v2/index.rst ✅
  - [x] docs/graphql/core/migrations/config-v2/manage-metadata.rst ✅
  - [x] docs/graphql/core/migrations/config-v2/migrations-setup.rst ✅
  - [x] docs/graphql/core/migrations/config-v2/upgrade-v2.rst ✅
  - [x] docs/graphql/core/migrations/config-v2/advanced/ ✅
    - [x] docs/graphql/core/migrations/config-v2/advanced/auto-apply-migrations.rst ✅
    - [x] docs/graphql/core/migrations/config-v2/advanced/index.rst ✅
    - [x] docs/graphql/core/migrations/config-v2/advanced/resetting-migrations.rst ✅
    - [x] docs/graphql/core/migrations/config-v2/advanced/rolling-back-migrations.rst ✅
    - [x] docs/graphql/core/migrations/config-v2/advanced/seed-data-migration.rst ✅
    - [x] docs/graphql/core/migrations/config-v2/advanced/writing-migrations-manually.rst ✅
  - [x] docs/graphql/core/migrations/config-v2/reference/ ✅
      - [x] docs/graphql/core/migrations/config-v2/reference/how-it-works.rst ✅
      - [x] docs/graphql/core/migrations/config-v2/reference/index.rst ✅
      - [x] docs/graphql/core/migrations/config-v2/reference/metadata-format.rst ✅
      - [x] docs/graphql/core/migrations/config-v2/reference/migration-file-format.rst ✅
- [x] docs/graphql/core/migrations/reference ✅
    - [x] docs/graphql/core/migrations/reference/how-it-works.rst ✅
    - [x] docs/graphql/core/migrations/reference/index.rst ✅
    - [x] docs/graphql/core/migrations/reference/metadata-format.rst ✅
    - [x] docs/graphql/core/migrations/reference/migration-file-format.rst ✅
</details>

<details>
<summary>docs/graphql/core/databases/bigquery/</summary>

  - [ ] docs/graphql/core/databases/bigquery/getting-started.rst
  - [ ] docs/graphql/core/databases/bigquery/index.rst
  - [ ] docs/graphql/core/databases/bigquery/schema
      - [ ] docs/graphql/core/databases/bigquery/schema/custom-field-names.rst
      - [ ] docs/graphql/core/databases/bigquery/schema/index.rst
</details>

<details>
<summary>docs/graphql/core/getting-started/</summary>

- [ ] docs/graphql/core/getting-started/docker-simple.rst
- [ ] docs/graphql/core/getting-started/first-graphql-query.rst
- [ ] docs/graphql/core/getting-started/index.rst
</details>

**NEW**

<details>
<summary>docs/graphql/core/databases/ </summary>

- [ ] docs/graphql/core/databases/connect-db.rst
- [ ] docs/graphql/core/databases/index.rst
- [ ] docs/graphql/core/databases/ms-sql-server/
- [ ] docs/graphql/core/databases/ms-sql-server/mutations
  - [ ] docs/graphql/core/databases/ms-sql-server/mutations/delete.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/mutations/index.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/mutations/insert.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/mutations/upsert.rst
- [ ] docs/graphql/core/databases/ms-sql-server/queries
  - [ ] docs/graphql/core/databases/ms-sql-server/queries/aggregation-queries.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/queries/index.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/queries/multiple-arguments.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/queries/multiple-queries.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/queries/nested-object-queries.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/queries/pagination.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/queries/performance.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/queries/query-filters.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/queries/simple-object-queries.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/queries/sorting.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/queries/variables-aliases-fragments-directives.rst
- [ ] docs/graphql/core/databases/ms-sql-server/schema
  - [ ] docs/graphql/core/databases/ms-sql-server/schema/custom-field-names.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/schema/tables.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/schema/using-existing-database.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/schema/views.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/schema/default-values
    - [ ] docs/graphql/core/databases/ms-sql-server/schema/default-values/index.rst
    - [ ] docs/graphql/core/databases/ms-sql-server/schema/default-values/mssql-column-presets.rst
    - [ ] docs/graphql/core/databases/ms-sql-server/schema/default-values/mssql-defaults.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/schema/table-relationships
    - [ ] docs/graphql/core/databases/ms-sql-server/schema/table-relationships/create.rst
    - [ ] docs/graphql/core/databases/ms-sql-server/schema/table-relationships/database-modelling.rst
    - [ ] docs/graphql/core/databases/ms-sql-server/schema/table-relationships/index.rst
    - [ ] docs/graphql/core/databases/ms-sql-server/schema/table-relationships/rename.rst
- [ ] docs/graphql/core/databases/ms-sql-server/subscriptions
  - [ ] docs/graphql/core/databases/ms-sql-server/subscriptions/index.rst
  - [ ] docs/graphql/core/databases/ms-sql-server/subscriptions/use-cases.rst
- [ ] docs/graphql/core/databases/postgres
  - [ ] docs/graphql/core/databases/postgres/index.rst
  - [ ] docs/graphql/core/databases/postgres/postgresql-types.rst
  - [ ] docs/graphql/core/databases/postgres/mutations
    - [ ] docs/graphql/core/databases/postgres/mutations/delete.rst
    - [ ] docs/graphql/core/databases/postgres/mutations/index.rst
    - [ ] docs/graphql/core/databases/postgres/mutations/insert.rst
    - [ ] docs/graphql/core/databases/postgres/mutations/multiple-mutations.rst
    - [ ] docs/graphql/core/databases/postgres/mutations/update.rst
    - [ ] docs/graphql/core/databases/postgres/mutations/upsert.rst
 
</details>

---
🚧 - Work in Progress
⁉️ - Work Done - Review Pending
⚠️ - Review Done - Changes Required
✅ - Review Done - Migration Complete

### CODE REVIEW

File: docs-new/docs/graphql/core/hasura-cli/hasura.mdx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    hasura [flags]
+
+## Options
+
+    --envfile string      .env filename to load ENV vars from (default ".env")
+    -h, --help                help for hasura
+    --log-level string    log level (DEBUG, INFO, WARN, ERROR, FATAL) (default "INFO")
+    --no-color            do not colorize output (default: false)
+    --project string      directory where commands are executed (default: current dir)
+    --skip-update-check   skip automatic update check on command execution
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/hasura-cli/hasura.mdx:32]: Looks like indentation is off for al

Please check in live docs to see how they look and probably wrap in (```) backticks for better readability as code blocks. Here and all other occurrences.

Screenshot 2022-01-19 at 7 11 03 PM

File: docs-new/docs/graphql/core/hasura-cli/hasura.mdx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## SEE ALSO
+
+- [hasura actions](/graphql/core/hasura-cli/hasura_actions) - Manage Hasura actions
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/hasura-cli/hasura.mdx:36]: Please use extension for internal links. Here and all other occurrences

**Review by KRRISH96 — CHANGES_REQUESTED** [review]
@rjitsu awesome work on the PR!

Here is some general feedback,
Most of the files appears to be code-block majority. So, Lets wrap them in triple backticks for avoiding any unintentional spacing/parse behaviour.

1. Wrap code blocks in backticks.
2. Fix indentation - refer atached screenshot in comment.
3. Use extension for all internal markdown links

**Comment by KRRISH96** [comment]
Screenshot 2022-01-19 at 8 34 49 PM

@rjitsu looks like the tabs are missed and the divs are not removed within content.

Please ensure those are removed after migrating the content.

File: docs-new/docs/graphql/core/hasura-cli/hasura.mdx (hunk 1 above)
Comment (rjitsu) [reply docs-new/docs/graphql/core/hasura-cli/hasura.mdx:32]: Fixed indentation and added backticks to code blocks.

File: docs-new/docs/graphql/core/hasura-cli/hasura.mdx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Synopsis
+
+    __
+     / /_   ____ _ _____ __  __ _____ ____ _
+    / __ \ / __ `// ___// / / // ___// __ `/
+    / / / // /_/ /(__  )/ /_/ // /   / /_/ /
+    /_/ /_/ \__,_//____/ \__,_//_/    \__,_/
+
+    hasura [flags]
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/hasura-cli/hasura.mdx:23]: These two needs to be separate code blocks and also remove extra indentation.

File: docs-new/docs/graphql/core/migrations/manage-metadata.mdx [hunk 4]
```diff
@@ -115,47 +94,30 @@ instance's metadata onto itself.
...
-
-</div>
-
-<div class="tab">
+<Tabs className="api-tabs">
+<TabItem value="cli" label="CLI">
+Metadata can be applied with the <Link to="/graphql/core/hasura-cli/hasura_metadata_apply">
+hasura metadata apply
+</Link> command.
+</TabItem>
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/migrations/manage-metadata.mdx:102]: Pls add an empty line before and after and starting the content from line beginning will help avoid using Link compinent.

The broken links with Link component wont be caught on build. SO lets use markdown links where ever possible.

File: docs-new/docs/graphql/core/migrations/manage-metadata.mdx [hunk 5]
```diff
@@ -171,70 +133,46 @@ request to replace the metadata. If an admin secret is set, add
...
-1.  Click on the settings (⚙) icon at the top right corner of the
-    console screen.
+<Tabs className="api-tabs">
 
-2.  Click on `Reload` button.
+<TabItem value="cli" label="CLI">
+Metadata can be reloaded with the <Link to="/graphql/core/hasura-cli/hasura_metadata_reload.mdx">hasura metadata reload</Link> command.
+</TabItem>
+<TabItem value="console" label="Console">
+1.Click on the settings (⚙) icon at the top right corner of the
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/migrations/manage-metadata.mdx:161]: please leave an empty space as first and last line when adding markdown with in the TabItem components.

Also, please add a space between 1.` `and the content for this and for all other usages

Comment (KRRISH96) [reply docs-new/docs/graphql/core/migrations/manage-metadata.mdx:161]: Cool. but please ensure an empty first line and last line is added for mdx content  inside the TabItem.

Else the content won't be parsed as markdown.

File: docs-new/docs/graphql/core/migrations/config-v2/migrations-setup.mdx [hunk 6]
```diff
@@ -193,18 +176,11 @@ git add .
...
 
-<div class="note">
-
-<div class="title">
-
-Note
+:::info Note
 
-</div>
+The version control set up should typically be done right after [Step 2](#step-2-set-up-a-project-directory)
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/migrations/config-v2/migrations-setup.mdx:181]: `migrations-project-init-v2` seems to be correct id from rst file. So, please use same ids and avoid the autogenerated ones from docusaurus.

Also, pls attach these ids to respective headers

File: docs-new/docs/graphql/core/migrations/config-v2/reference/metadata-format.mdx [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+- [actions.graphql](#actionsgraphql)
+- [actions.yaml](#actionsyaml)
+- [allow_list.yaml](#allow_listyaml)
+- [cron_triggers.yaml](#cron_triggersyaml)
+- [query_collections.yaml](#query_collectionsyaml)
+- [remote_schemas.yaml](#remote_schemasyaml)
+- [version.yaml](#versionyaml)
+- [databases](#databases)
+- [databases.yaml](#databasesyaml)
+- [public_t1.yaml](#public_t1yaml)
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/migrations/config-v2/reference/metadata-format.mdx:39]: please add custom ids for all the respective headers

Please avoid relying on auto generated header ids - they are prone to break if header text changes

File: docs-new/docs/graphql/core/migrations/config-v2/migrations-setup.mdx (hunk 6 above)
Comment (KRRISH96) [inline docs-new/docs/graphql/core/migrations/config-v2/migrations-setup.mdx:181]: hash-param (id) mismatch - `migrations-project-init-v2`

File: docs-new/docs/graphql/core/migrations/config-v2/migrations-setup.mdx [hunk 8]
```diff
@@ -284,18 +254,11 @@ git add .
...
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/migrations/config-v2/migrations-setup.mdx:259]: hash-param (id) mismatch - `migrations-project-init-v2`

File: docs-new/docs/graphql/core/migrations/config-v2/reference/metadata-format.mdx [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+
+- [actions.graphql](#actionsgraphql)
+- [actions.yaml](#actionsyaml)
+- [allow_list.yaml](#allow_listyaml)
+- [cron_triggers.yaml](#cron_triggersyaml)
+- [functions.yaml](#functions-yaml)
+- [query_collections.yaml](#query_collectionsyaml)
+- [remote_schemas.yaml](#remote_schemasyaml)
+- [tables.yaml](#tables-yaml)
+- [version.yaml](#versionyaml)
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/migrations/config-v2/reference/metadata-format.mdx:38]: These links are actually table of contents (TOC) for the sub section.

Please use - `TOCInline` component which will auto generate the subsection link list.

The below should work for this doc.
```jsx
<TOCInline toc={toc[1].children} />
```

Please refer the [Sub-Subitem section in guide](https://hge-docs-migration.netlify.app/migration-guide/table-of-contents#sub-subitem)

Also no need to attach custom id for section headers when its not added in original file.

File: docs-new/docs/graphql/core/migrations/config-v2/reference/metadata-format.mdx [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+
+:::info Note
+
+The output of the [export_metadata](/graphql/core/api-reference/metadata-api/manage-metadata.mdx) metadata API is a JSON
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/migrations/config-v2/reference/metadata-format.mdx:42]: missing hash-param - this links to a sun section in the linked doc, pls add hash param for correct section (`#metadata-export-metadata`)

**Review by KRRISH96 — COMMENTED** [review]
Along with mentioned comments. Please recheck the sidebar positions and labels.

File: docs-new/docs/graphql/core/migrations/config-v2/manage-metadata.mdx [hunk 11]
```diff
@@ -176,7 +178,7 @@ Metadata can be reloaded with the [hasura metadata reload](/graphql/core/hasura-
 
 <TabItem value="api" label="API">
 
-The reload of metadata can be done via the `metadata_reload_metadata`
+The reload of metadata can be done via the reload_metadata](/graphql/core/api-reference/metadata-api/manage-metadata.mdx#metadata-reload-metadata)
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/migrations/config-v2/manage-metadata.mdx:181]: missing opening `[` for link syntax

File: docs-new/docs/graphql/core/migrations/config-v2/advanced/auto-apply-migrations.mdx [hunk 12]
```diff
@@ -1,17 +1,18 @@
-<div class="meta"
-description="Auto-apply migrations and metadata when the server starts"
-keywords="hasura, docs, auto-apply, migration, metadata, server">
-
-</div>
+---
+sidebar_label: Auto-apply migrations/metadata (config v2)
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/migrations/config-v2/advanced/auto-apply-migrations.mdx:2]: minor mismatch in sidebar label

`Auto-apply migrations/metadata when the server starts (config v2)`

File: docs-new/docs/graphql/core/migrations/advanced/auto-apply-migrations.mdx [hunk 13]
```diff
@@ -1,17 +1,18 @@
-<div class="meta"
-description="Auto-apply migrations and metadata when the server starts"
-keywords="hasura, docs, auto-apply, migration, metadata, server">
-
-</div>
+---
+sidebar_label: Auto-apply migrations/metadata
```
Comment (KRRISH96) [inline docs-new/docs/graphql/core/migrations/advanced/auto-apply-migrations.mdx:2]: Minor mismatch in sidebar_label

`Auto-apply migrations/metadata when the server starts`

**Review by KRRISH96 — COMMENTED** [review]
3 minor changes and migrations directory is GTG

**Pull request merged**